src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -437,7 +437,15 @@ bool HandleRedirect (HttpStatusCode redirectCode, HttpURLConnection httpConnecti
 			if (redirectState.RedirectCounter >= MaxAutomaticRedirections)
 				throw new WebException ($"Maximum automatic redirections exceeded (allowed {MaxAutomaticRedirections}, redirected {redirectState.RedirectCounter} times)");
 
-			Uri location = new Uri (locationHeader [0], UriKind.Absolute);
+			string redirectUrl = locationHeader[0];
+			string protocol = httpConnection.URL?.Protocol;
+			if (redirectUrl.StartsWith("//") && protocol != null && protocol.StartsWith("http", StringComparison.OrdinalIgnoreCase))
```

Comment:
The commit message says:

> This commit adds support for redirecting to urls that do not specify a scheme. In such cases, the scheme of the previous request is used.

The actual change only uses the previous scheme if the pervious protocol starts with `http`, which is counter to the commit message.

Is it desirable to restrict this to just `http` and `https`, or should it do as the commit message says and use the previous protocol, no matter what it is?

If it should restrict to just `http*`, then the commit message should say so.